src/app/(nextjs_migration)/(guest)/breach-details/[breachName]/page.tsx

```diff
@@ -0,0 +1,368 @@
+    };
+  } else if (categoryId === "sensitive-breach") {
+    return {
+      subhead: l10n.getString("sensitive-sites"),
+      body: l10n.getString("sensitive-sites-copy"),
+    };
+  } else {
+    return {
+      subhead: l10n.getString("what-is-a-website-breach"),
+      body: l10n.getString("website-breach-blurb"),
+    };
+  }
+}
+
+function makeBreachDetail(breach: ReturnType<typeof getBreachCategory>) {
+  const breachDetail = getBreachDetail(breach);
+  return (
+    <>
+      <h2>{breachDetail.subhead}</h2>
+      <div>{breachDetail.body}</div>
+    </>
+  );
+}
+
+function getBreachCategory(breach: Breach) {
+  if (
+    [
+      "Exactis",
+      "Apollo",
+      "YouveBeenScraped",
+      "ElasticsearchSalesLeads",
+      "Estonia",
+      "MasterDeeds",
+      "PDL",
+    ].includes(breach.Name)
+  ) {
+    return "data-aggregator-breach";
+  }
+  if (breach.IsSensitive) {
+    return "sensitive-breach";
+  }
+  if (breach.Domain !== "") {
+    return "website-breach";
+  }
+  return "data-aggregator-breach";
+}
+
+function compareBreachDates(breach: Breach) {
+  const breachDate = new Date(breach.BreachDate);
+  const addedDate = new Date(breach.AddedDate);
+  const timeDiff = Math.abs(breachDate.getTime() - addedDate.getTime());
+  const dayDiff = Math.ceil(timeDiff / (1000 * 3600 * 24));
+  if (dayDiff > 90) {
+    return true;
+  }
+  return false;
+}
+function getSortedDataClasses(
+  breach: Breach,
+  isUserBrowserFirefox = false,
+  isUserLocaleEnUs = false,
+  isUserLocalEn = false,
+  changePWLink = false
+) {
+  const l10n = getL10n();
+  const priorityDataClasses: any = getAllPriorityDataClasses(
+    isUserBrowserFirefox,
+    isUserLocaleEnUs,
+    changePWLink
+  );
+
+  const sortedDataClasses = {
+    priority: [] as any[],
+    lowerPriority: [] as any[],
+  };
+
+  const dataClasses: any[] = breach.DataClasses;
+  dataClasses.forEach((dataClass) => {
+    const dataType = l10n.getString(dataClass);
+    if (priorityDataClasses[dataClass]) {
+      priorityDataClasses[dataClass].dataType = dataType;
+      sortedDataClasses.priority.push(priorityDataClasses[dataClass]);
+      return;
+    }
+    sortedDataClasses.lowerPriority.push(dataType);
+  });
+  sortedDataClasses.priority.sort((a, b) => {
+    return b.weight - a.weight;
+  });
+  sortedDataClasses.lowerPriority = sortedDataClasses.lowerPriority.join(
+    ", "
+  ) as any;
+  return sortedDataClasses;
+}
+
+function makeDataSection(breach: Breach) {
+  const dataClasses = getSortedDataClasses(breach);
+
+  const output = dataClasses.priority.map((dataClass, dataIndex) => (
+    <li key={`data-class-${dataClass.glyphName}`}>
+      <Image src={glyphs[dataClass.glyphName]} width="24" alt="" />
+      {dataClass.dataType}
+    </li>
+  ));
+
+  const lastOutputItem = (
+    <li>
+      <Image alt="" src={MoreIcon} />
+      {dataClasses.lowerPriority}
+    </li>
+  );
+
+  return (
+    <>
+      {output} {lastOutputItem}
```

Comment:
https://monitor.firefox.com/breach-details/126

In fact, I found 69/673 (10.25%) breaches without "more" data classes.

```js
import CachedFetch from "@11ty/eleventy-fetch";
import * as cheerio from "cheerio";

const noMas = [];
const breaches = await getBreaches();
for (const breach of breaches) {
  const $ = await getBreachPage(breach.Name);
  const more = $("ul.breach-detail-compromised-list li:last-of-type").text().trim();
  if (!more) {
    noMas.push(breach.Name);
  }
}
console.log(`${noMas.length}/${breaches.length} breaches without "more" data classes`);
console.log(noMas);

async function getBreaches() {
  return CachedFetch("https://haveibeenpwned.com/api/v3/breaches", {type: "json", duration: "1d"});
}

async function getBreachPage(breachName) {
  const res = await CachedFetch(`https://monitor.firefox.com/breach-details/${breachName}`, { type: "text", duration: "1d" });
  return cheerio.load(res);
}
```

Obviously, not a blocker since this is already in production.

Reply:
Of course you did :smile: Well, might as well fix it now: d6367ef79

![image](https://github.com/mozilla/blurts-server/assets/4251/78c96582-229a-40d1-b9a6-25afe92753de)